File: Solutions/C#/Accepted/70-climbing-stairs.cs
/*
Problem Title: 70 - Climbing Stairs
Lang: C#
Runtime: 36 ms
Memory: 14.6 MB
Problem Url: https://leetcode.com/problems/climbing-stairs
Submission Url: https://leetcode.com/problems/climbing-stairs/submissions/302035652
*/

public class Solution {
    public int ClimbStairs(int n) {
        int res = 1, last = 1, second_last = 1;
        for(int step = 2; step <= n; step ++){
            res = last + second_last;
            second_last = last;
            last = res;
        }
        return res;
    }
}

File: Solutions/C++/Accepted/70-climbing-stairs.cpp
/*
Problem Title: 70 - Climbing Stairs
Lang: C++
Runtime: 2 ms
Memory: 6.2 MB
Problem Url: https://leetcode.com/problems/climbing-stairs
Submission Url: https://leetcode.com/problems/climbing-stairs/submissions/772314959
*/

class Solution {
public:
    int climbStairs(int n) {
        vector<int> dp(n+2, 0);
        dp[1] = 1;
        dp[2] = 2;
        for(int i = 3; i <= n; i++) {
            dp[i] = dp[i-1] + dp[i-2];
        }
        return dp[n];
    }
};